UnitsNet/CustomCode/Quantities/Duration.extra.cs

```diff
@@ -18,7 +18,7 @@ public TimeSpan ToTimeSpan()
             if ( Seconds > TimeSpan.MaxValue.TotalSeconds ||
                 Seconds < TimeSpan.MinValue.TotalSeconds )
                 throw new ArgumentOutOfRangeException( nameof( Duration ), "The duration is too large or small to fit in a TimeSpan" );
-            return TimeSpan.FromSeconds( Seconds );
+            return TimeSpan.FromTicks((long)(Seconds * TimeSpan.TicksPerSecond));
```

Comment:
Oh wow, I never realized FromSeconds [rounded to milliseconds](https://docs.microsoft.com/en-us/dotnet/api/system.timespan.fromseconds?view=net-6.0). Today I learned!

Reply:
Yeah, it really depends on the framework you're using.

.NET 5/6 does not round, it uses exact ticks ([see](https://github.com/dotnet/runtime/blob/6019cdace12ec9bc5117e515b9a6fccfd5708bde/src/libraries/System.Private.CoreLib/src/System/TimeSpan.cs#L215)):
```
private static TimeSpan Interval(double value, double scale)
{
	if (double.IsNaN(value))
		ThrowHelper.ThrowArgumentException_Arg_CannotBeNaN();
	return IntervalFromDoubleTicks(value * scale);
}

private static TimeSpan IntervalFromDoubleTicks(double ticks)
{
	if ((ticks > long.MaxValue) || (ticks < long.MinValue) || double.IsNaN(ticks))
		ThrowHelper.ThrowOverflowException_TimeSpanTooLong();
	if (ticks == long.MaxValue)
		return TimeSpan.MaxValue;
	return new TimeSpan((long)ticks);
}
```

In .NET Framework 4.8, the values are rounded to milliseconds ([see](https://referencesource.microsoft.com/#mscorlib/system/timespan.cs,228)):
```
private static TimeSpan Interval(double value, int scale) {
	if (Double.IsNaN(value))
		throw new ArgumentException(Environment.GetResourceString("Arg_CannotBeNaN"));
	Contract.EndContractBlock();
	double tmp = value * scale;
	double millis = tmp + (value >= 0? 0.5: -0.5);
	if ((millis > Int64.MaxValue / TicksPerMillisecond) || (millis < Int64.MinValue / TicksPerMillisecond))
		throw new OverflowException(Environment.GetResourceString("Overflow_TimeSpanTooLong"));
	return new TimeSpan((long)millis * TicksPerMillisecond);
}
```

Mono also rounds ([see](https://github.com/mono/mono/blob/4b5e2591eb299eb34d60b5217b14c1ad9f11e238/mcs/class/referencesource/mscorlib/system/timespan.cs#L228)), its implementation is the same as in the .NET Framework.

So you can't rely on its accuracy, that's why I've resorted to custom TimeSpanExact.FromXXX methods which use FromTicks internally.